src/main/java/htsjdk/variant/variantcontext/writer/VariantContextWriterBuilder.java

```diff
@@ -468,68 +486,75 @@ else if (STREAM_TYPES.contains(this.outType))
      * attempt to resolve any symlinks and try again.  If that fails, and the output file exists
      * but is neither a file or directory then VCF_STREAM is returned.
      */
-    protected static OutputType determineOutputTypeFromFile(final File f) {
-        if (isBCF(f)) {
+    protected static OutputType determineOutputTypeFromFile(final Path path) {
+        if (isBCF(path)) {
             return OutputType.BCF;
-        } else if (isCompressedVCF(f)) {
+        } else if (isCompressedVCF(path)) {
             return OutputType.BLOCK_COMPRESSED_VCF;
-        } else if (isVCF(f)) {
+        } else if (isVCF(path)) {
             return OutputType.VCF;
         }
         else {
             // See if we have a special file (device, named pipe, etc.)
-            final File canonical = new File(IOUtil.getFullCanonicalPath(f));
-            if (!canonical.equals(f)) {
-                return determineOutputTypeFromFile(canonical);
+            try {
+                // Try to resolve symlinks etc.
+                File file = path.toFile();
+                final File canonical = new File(IOUtil.getFullCanonicalPath(file));
+                if (!canonical.equals(file)) {
+                    return determineOutputTypeFromFile(IOUtil.toPath(canonical));
+                }
+            } catch (UnsupportedOperationException x) {
+                // Thrown for paths that are not from the default provider
+                // those are not going to have symlinks anyways
             }
-            else if (f.exists() && !f.isFile() && !f.isDirectory()) {
+            if (Files.exists(path) && !Files.isRegularFile(path) && !Files.isDirectory(path)) {
                 return OutputType.VCF_STREAM;
             } else {
                 return OutputType.UNSPECIFIED;
             }
         }
     }
 
-    private static boolean isVCF(final File outFile) {
-        return outFile != null && outFile.getName().endsWith(IOUtil.VCF_FILE_EXTENSION);
+    private static boolean isVCF(final Path outPath) {
+        return outPath != null && outPath.toString().endsWith(IOUtil.VCF_FILE_EXTENSION);
```

Comment:
I am working actually (although I did not have time to continue) in an HTTP/S filesystem, and I am open to opinions for how to handle that and being useful for the htsjdk codebase - my first idea was to get rid of the query strings in the URLs for both `getFileName` and `toString`, but it might change - for sure it will be passed to the `toUri` method.

Reply:
If you are interested, look into the https://github.com/magicDGS/jsr203-http project (as I said, I am planning to come back to it soon, but I had some problems with my personal computer which was a pain for my personal projects...)

Reply:
@magicDGS this provider looks very cool! This could make it easy for Java programs to read from [signed URLs](https://cloud.google.com/storage/docs/access-control/signed-urls) that point to files in buckets (or elsewhere, of course).

Reply:
@lbergelson adding a `.getFileName`, though I still need a `.toString` after since it returns a `Path`.